### DISCUSSION TITLE
Protocol command to set VTX channel, pwr, pitmode.

### DISCUSSION THREAD

**Discussion by JimB40** [ExpressLRS/ExpressLRS#2181] [Ideas]
Idea came while racing whoops. ELRS due to AIO FC is most popular in whoop racing scene.
During race day with 40 participants everything is time sentesive (happens fast) and time slots to configure whoop is very limited.
1-2 minutes between races max, when pilots can do any config. Additional obstacle is that AIO switches on VTX when usb is connected.

So currently to change VTX params we have 3 ways: OSD, ELRS script, MSP command via LUA script

1. OSD - radio & googles need to be powered and set to proper channel. Time to change 10-30 secs - Browsing takes 5-6 steps 
Nobody else has to be on the same channel otherwise. If it happens pilot has to find this other pilot and ask to disconnect causing delays. This is PIA specially when two racing pilots have to exchange channels due to interferences. 
2. ELRS script - only radio powered- Time 10-30 secs depending how fast parameters are read and how fast pilot is using script browsing. Plus is that changing channel on radio will sync backpack  
3. MSP command via LUA script - only radio powered, Setting VTX via ELRS script needs to be off - Dedicated LUA script only with race channels. Time 5 sec. Drawback is backpack wil not be synced.

So idea is to have new protocol command that will set VTX channel, PWR and pitmode
Dedicated LUA script with racing channels will send this command via crossfireTelemetryPush() to ELRS module, then everything happens like ELRS script VTX Admin Send command was issued.

This will allow to
- quickly (1-2 sec)select racing channel even googles are on different channel
- quickly free racing channel either by selecting not used channel or pitmode

For safety reasons this command may be executed while not armed.

**Comment by JyeSmith** [comment]
Sounds like you want a EdgeTX widget that is readily available on the home screen.  But this will only be available to some handsets like the TX16.

There are better ways to do this via a RaceBackpack that is communicating with the timing firmware.  It will auto send out any channel changes to ELRS Backpacks and the user does not need to do anything or even think about changes.  Work in this space is being progressed by the RotorHazard guys.

**Reply by JimB40** [reply]
Yes. Regarding LUA script. It may be color LCD radio widget or BW radio telemetry script. Both quickly accessible and covering all types of handsets.

Well RH dev will take some time. Checked OSD Fusion implementation to display RH info and there is so far caveat as MAC addres has to be registered. This would take more than hour with 40 pilots and it's prone to errors as MAC has to be entered manually. Maybe there will be better solution but so far too much to fit in to 8h of Race Day. 

There is also one more advadange of protocol command versus MSP. MSP needs parser that is quite large for BW handsets with limited memory, while protocol command would need only one LUA code line. Rest including broadcasting to backpacks will be done by ELRS module FW.

Quick and efficient.
As Pareto says 20% of features are used 80% of time and changing VTX settings is one of them.
